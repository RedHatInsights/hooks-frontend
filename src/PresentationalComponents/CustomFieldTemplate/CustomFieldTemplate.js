import React from 'react';
import PropTypes from 'prop-types';

export const CustomFieldTemplate = ({ help, description, errors, children }) => {
    return (
        <div>
            { children }
            { description }
            { errors }
            { help }
        </div>
    );
};

CustomFieldTemplate.propTypes = {
    help: PropTypes.object,
    description: PropTypes.object,
    errors: PropTypes.object,
    children: PropTypes.array
};
