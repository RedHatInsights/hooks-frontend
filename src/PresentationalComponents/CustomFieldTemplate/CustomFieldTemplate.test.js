import toJson from 'enzyme-to-json';

import { CustomFieldTemplate } from './CustomFieldTemplate';

describe('CustomFieldTemplate', () => {
    it('expect to render', () => {
        const testProps = {
            children: [ <div key="test-div"></div> ],
            description: <span/>,
            errors: <div key="error-test-div"></div>
        };

        const wrapper = shallow(
            <CustomFieldTemplate { ...testProps } />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
