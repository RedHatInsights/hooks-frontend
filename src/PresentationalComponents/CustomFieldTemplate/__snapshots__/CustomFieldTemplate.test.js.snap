// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomFieldTemplate expect to render 1`] = `
<div>
  <div
    key="test-div"
  />
  <span />
  <div
    key="error-test-div"
  />
</div>
`;
